packages/flutter/test/widgets/sensitive_content_test.dart

```diff
@@ -0,0 +1,1414 @@
+    WidgetTester tester,
+  ) async {
+    await tester.pumpWidget(
+      SensitiveContent(sensitivityLevel: ContentSensitivity.sensitive, child: Container()),
+    );
+
+    expect(
+      sensitiveContentSetting.getContentSenstivityState()!.currentContentSensitivitySetting,
+      equals(ContentSensitivity.sensitive),
+    );
+    expect(sensitiveContentSetting.getContentSenstivityState()!.sensitiveWidgetCount, equals(1));
+    expect(setContentSensitivityCallCount, 1);
+  });
+
+  testWidgets(
+    'disposing only SensitiveContent widget in the tree sets content sensitivity back to the default as expected',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        SensitiveContent(sensitivityLevel: ContentSensitivity.sensitive, child: Container()),
+      );
+      await tester.pumpWidget(Container());
+
+      expect(
+        sensitiveContentSetting.getContentSenstivityState()!.currentContentSensitivitySetting,
+        equals(defaultContentSensitivitySetting),
+      );
+      expect(sensitiveContentSetting.getContentSenstivityState()!.sensitiveWidgetCount, equals(0));
+    },
+  );
+
+  group(
+    'one sensitive SensitiveContent widget in the tree determines content sensitivity for tree as expected',
+    () {
+      // Tests with other sensitive widget(s):
+      testWidgets('with another sensitive widget', (WidgetTester tester) async {
+        await tester.pumpWidget(
+          Column(
+            children: <Widget>[
+              SensitiveContent(sensitivityLevel: ContentSensitivity.sensitive, child: Container()),
+              SensitiveContent(sensitivityLevel: ContentSensitivity.sensitive, child: Container()),
+            ],
+          ),
+        );
+
+        expect(
+          sensitiveContentSetting.getContentSenstivityState()!.currentContentSensitivitySetting,
+          equals(ContentSensitivity.sensitive),
+        );
+        expect(
+          sensitiveContentSetting.getContentSenstivityState()!.sensitiveWidgetCount,
+          equals(2),
+        );
+        expect(setContentSensitivityCallCount, 1);
+      });
+
+      testWidgets('when it gets disposed with another sensitive widget', (
+        WidgetTester tester,
+      ) async {
+        const Key sc1Key = Key('sc1');
+        final DisposeTester sc1 = DisposeTester(
+          child: SensitiveContent(
+            key: sc1Key,
+            sensitivityLevel: ContentSensitivity.sensitive,
+            child: Container(),
+          ),
+        );
+        final SensitiveContent sc2 = SensitiveContent(
+          sensitivityLevel: ContentSensitivity.sensitive,
+          child: Container(),
+        );
+
+        await tester.pumpWidget(Column(children: <Widget>[sc1, sc2]));
+
+        final DisposeTesterState sc1DiposeTesterState = tester.firstState<DisposeTesterState>(
+          find.byKey(sc1Key),
+        );
+        sc1DiposeTesterState.disposeWidget();
+        await tester.pumpAndSettle();
```

Comment:
Would this still work with just `pump`? I would do that if you can. Here and below there are a bunch more like it.